Makefile

```diff
@@ -0,0 +1,126 @@
+# Include my.env and export it so variables set in there are available
+# in the Makefile.
+include my.env
+export
+
+# Set these in the environment to override them. This is helpful for
+# development if you have file ownership problems because the user
+# in the container doesn't match the user on your host.
+ICHNAEA_UID ?= 10001
+ICHNAEA_GID ?= 10001
+
+# Set this in the environment to force --no-cache docker builds.
+DOCKER_BUILD_OPTS :=
+ifeq (1, ${NOCACHE})
+DOCKER_BUILD_OPTS := --no-cache
+endif
+
+DC := $(shell which docker-compose)
+
+.PHONY: help
+help: default
+
+.PHONY: default
+default:
+	@echo "Usage: make RULE"
+	@echo ""
+	@echo "Ichnaea make rules:"
+	@echo ""
+	@echo "  build            - build docker containers"
+	@echo "  setup            - drop and recreate service state"
+	@echo "  run              - run webapp"
+	@echo "  runcelery        - run scheduler and worker"
+	@echo "  runservices      - run service containers (mysql, redis, etc)"
+	@echo "  stop             - stop all service containers"
+	@echo ""
+	@echo "  shell            - open a shell in the app container"
+	@echo "  mysql            - open mysql prompt"
+	@echo "  clean            - remove all build, test, coverage and Python artifacts"
+	@echo "  lint             - lint code"
+	@echo "  test             - run unit tests"
+	@echo "  testshell        - open a shell in the test environment"
+	@echo "  docs             - generate Sphinx HTML documentation, including API docs"
+	@echo "  buildjs          - generate js static assets"
+	@echo "  buildcss         - generate css static assets"
+	@echo ""
+	@echo "  help             - see this text"
+	@echo ""
+	@echo "See https://mozilla.github.io/ichnaea/ for more documentation."
+
+.PHONY: clean
+clean:
+	rm .docker-build* || true
+
+my.env:
+	@if [ ! -f my.env ]; \
+	then \
+	echo "Copying my.env.dist to my.env..."; \
+	cp docker/config/my.env.dist my.env; \
+	fi
+
+.docker-build:
+	make build
+
+.PHONY: build
+build: my.env
+	${DC} build ${DOCKER_BUILD_OPTS} \
+	    --build-arg userid=${ICHNAEA_UID} \
+	    --build-arg groupid=${ICHNAEA_GID} \
+	    node app
+	touch .docker-build
+
+.PHONY: setup
+setup: my.env
+	${DC} run app shell ./docker/run_setup.sh
+
+.PHONY: shell
+shell: my.env .docker-build
+	${DC} run --rm app shell
+
+.PHONY: mysql
+mysql: my.env .docker-build
+	${DC} up -d mysql
+	@echo "=============================="
+	@echo "Root password is \"location\"."
+	@echo "=============================="
+	${DC} exec mysql mysql --user root --password
```

Comment:
How about adding the password and automatically connecting to the `location` database?

```
${DC} exec mysql mysql --user root --password=location location
```

Reply:
Aha! I couldn't get `mysql --user root --password location` working--seemed like `--password` was treated as an indicator to ask for the password so I did this janky thing.

But it works with `--password=location` and that's awesome! I'll fix it.